src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -960,8 +960,19 @@ else if (conf.commitlog_segment_size.toKibibytes() < 2 * conf.max_mutation_size.
         if (conf.client_encryption_options != null)
             conf.client_encryption_options.applyConfig();
 
-        if (conf.jmx_encryption_options != null)
-            conf.jmx_encryption_options.applyConfig();
+        if (conf.jmx_server_options == null)
+        {
+            conf.jmx_server_options = JMXServerOptions.createParsingSystemProperties();
+        }
+        else if (JMXServerOptions.isEnabledBySystemProperties())
+        {
+            throw new ConfigurationException("Configure either jmx_server_options in cassandra.yaml and comment out " +
```

Comment:
what if we set at both places, do we wanna throw exception vs. log a error/warning message, considering the yaml have higher precedence than the cassandra-env.sh?

Reply:
I think this is already covered. 

`if (conf.jmx_server_options == null)` - this means that if it is configured in yaml, this will _not_ be null, right?

So then it this `if` evaluates to `false`, then it goes to `else if` and 

`else if (JMXServerOptions.isEnabledBySystemProperties())`

if this is true, then it throws.

So when does it throw? When we have non-null yaml setting and when properties are set - that means it is set at the both places, so we throw.